### PR TITLE
Migrate Thailand holidays to holiday groups

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Migrates Thailand holidays to `holiday_group` standard.

This PR also implements #1106's new localization test methods and expands the loc's test case to cover all of the 2022 holidays.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/thailand.py [hunk 1]
```diff
@@ -684,12 +713,21 @@ def _add_with_observed(dt, holiday_name) -> None:
         }
         # For years with exact date data.
         if year in raeknakhwan_dates:
-            _add_with_observed(
-                date(year, *raeknakhwan_dates[year]), raeknakhwan
+            _add_observed(
+                self._add_holiday(
+                    raeknakhwan,
+                    date(year, *raeknakhwan_dates[year]),
```
Comment (KJhellico) [inline holidays/countries/thailand.py:719]:
```suggestion
                    *raeknakhwan_dates[year],
```

Comment (PPsyrius) [reply holidays/countries/thailand.py:719]: Thanks for the suggestion, implemented :)

**Comment by KJhellico** [comment]
@PPsyrius, look at my PR to your repo, please. ;)

**Review by arkid15r — COMMENTED** [review]
It seems that l10n files need to be updated too.
Otherwise looks good, nicely done!

File: holidays/countries/thailand.py [hunk 2]
```diff
@@ -450,19 +459,16 @@ def _add_with_observed(dt, holiday_name) -> None:
...
 
-        if 1976 <= year <= 2016:
-            _add_with_observed(
-                date(year, AUG, 12),
-                tr(
+        if year >= 1976:
+            name = (
+                tr("วันเฉลิมพระชนมพรรษาสมเด็จพระบรมราชชนนีพันปีหลวง")
+                if year >= 2017
+                else tr(
```
[the remark this answers is not in the document]
Comment (PPsyrius) [reply holidays/countries/thailand.py:466]: I'm afraid that's not the case:
```python
        # HM Queen Sirikit the Queen Mother's Birthday.
        # วันเฉลิมพระชนมพรรษา พระบรมราชินีนาถ ( 1976-2017)
        # วันเฉลิมพระชนมพรรษา พระบรมราชชนนีพันปีหลวง (2017-Present)
        # Status: In-Use.
        # Started in 1976 (B.E. 2519) alongside Mother's Day.
        # Initial celebration as HM Queen Sirikit's Birthday.
        # Now acts as the Queen Mother from 2017 onwards.

        if year >= 1976:
            name = (
                tr("วันเฉลิมพระชนมพรรษาสมเด็จพระบรมราชชนนีพันปีหลวง")
                if year >= 2017
                else tr(
                    "วันเฉลิมพระชนมพรรษาสมเด็จพระนางเจ้าสิริกิติ์ "
                    "พระบรมราชินีนาถ"
                )
            )
            _add_observed(self._add_holiday(name, AUG, 12))
```

[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/countries/thailand.py:466]: It's not indeed. Somebody needs to adjust his rulers.

**Pull request merged**